framework/src/service/ServiceListeners.cpp

```diff
@@ -499,10 +499,10 @@ void ServiceListeners::AddToSet_unlocked(
   int cache_ix,
   const std::string& val)
 {
-  std::list<ServiceListenerEntry>& l = cache[cache_ix][val];
+  std::set<ServiceListenerEntry>& l = cache[cache_ix][val];
   if (!l.empty()) {
 
-    for (std::list<ServiceListenerEntry>::const_iterator entry = l.begin();
+    for (std::set<ServiceListenerEntry>::const_iterator entry = l.begin();
```

Comment:
Should this be a range-based for loop? Something like:

for (auto const entry : l) { if (receivers.count(entry)) set.insert(entry); }

?